utils/build-script

```diff
@@ -1221,42 +1221,22 @@ details of the setups of other systems or automated environments.""")
              "targets to build the Swift standard library for, or 'all'.",
         type=arguments.type.shell_split, default=["all"])
 
+    # Add options to select projects to build.
     projects_group = parser.add_argument_group(
         title="Options to select projects")
-    projects_group.add_argument(
-        "-l", "--lldb",
-        help="build LLDB",
-        action="store_true",
-        dest="build_lldb")
-    projects_group.add_argument(
-        "-b", "--llbuild",
-        help="build llbuild",
-        action="store_true",
-        dest="build_llbuild")
-    projects_group.add_argument(
-        "-p", "--swiftpm",
-        help="build swiftpm",
-        action="store_true",
-        dest="build_swiftpm")
-    projects_group.add_argument(
-        "--xctest",
-        help="build xctest",
-        action=arguments.action.optional_bool,
-        dest="build_xctest")
-    projects_group.add_argument(
-        "--foundation",
-        help="build foundation",
-        action=arguments.action.optional_bool,
-        dest="build_foundation")
-    projects_group.add_argument(
-        "--libdispatch",
-        help="build libdispatch",
-        action=arguments.action.optional_bool,
-        dest="build_libdispatch")
-    projects_group.add_argument(
-        "--build-ninja",
-        help="build the Ninja tool",
-        action=arguments.action.optional_bool)
+    for cls in products.known_product_classes:
+        # Special case for short form aliases.
+        short_alias = {products.LLDB: "l",
+                       products.LLBuild: "b",
+                       products.SwiftPM: "p"}.get(cls)
+        argnames = []
+        if short_alias is not None:
+            argnames.append("-" + short_alias)
+        argnames.append("--" + cls.product_name())
+        projects_group.add_argument(
+            *argnames, help="enable {}".format(cls.__name__),
+            action=arguments.action.optional_bool,
+            dest="products.{}.build".format(cls.product_name()))
```

Comment:
We cannot use `optional_bool` here. https://github.com/apple/swift/pull/2983#discussion_r67276355